File: articles/inequality-aversion-economic-decision-making/fortran/inequality_aversion_utility.f90
program inequality_aversion_utility
  implicit none
  real :: self_payoff, other_payoff, alpha, beta, utility

  self_payoff = 0.30
  other_payoff = 0.70
  alpha = 1.5
  beta = 0.6

  utility = self_payoff - alpha * max(other_payoff - self_payoff, 0.0) - beta * max(self_payoff - other_payoff, 0.0)

  print *, "Fehr-Schmidt utility:", utility
end program inequality_aversion_utility

File: articles/inequality-aversion-economic-decision-making/c/inequality_aversion_utility.c
#include <stdio.h>

double max_double(double a, double b) {
    return a > b ? a : b;
}

double fehr_schmidt(double self_payoff, double other_payoff, double alpha, double beta) {
    return self_payoff
           - alpha * max_double(other_payoff - self_payoff, 0.0)
           - beta * max_double(self_payoff - other_payoff, 0.0);
}

int main(void) {
    printf("Fehr-Schmidt utility: %.3f\n", fehr_schmidt(0.30, 0.70, 1.5, 0.6));
    return 0;
}
